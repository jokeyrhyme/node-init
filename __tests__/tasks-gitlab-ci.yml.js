/* @flow */
'use strict';

const {
  isNeeded,
  lib: { editConfig }
} = require('../lib/tasks/gitlab-ci.yml.js');

test('editConfig({}, { hasYarn: false })', () => {
  const results = editConfig({}, { hasYarn: false, majors: [8] });
  expect(results).toMatchSnapshot();
});
test('editConfig({}, { hasYarn: true })', () => {
  const results = editConfig({}, { hasYarn: true, majors: [8] });
  expect(results).toMatchSnapshot();
});

test('isNeeded()', async () => {
  const result = await isNeeded({ cwd: process.cwd(), scope: '' });
  expect(typeof result).toBe('boolean');
});
