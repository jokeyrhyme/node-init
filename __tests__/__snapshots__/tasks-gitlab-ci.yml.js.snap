// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editConfig({}, { hasYarn: false }) 1`] = `
Object {
  "before_script": Array [
    "apt-get update -qq",
    "apt-get install -qy libelf1",
    "npm install --global npm",
    "npm install",
  ],
  "test": Object {
    "image": "node:8",
    "script": Array [
      "npm test",
    ],
  },
}
`;

exports[`editConfig({}, { hasYarn: true }) 1`] = `
Object {
  "before_script": Array [
    "apt-get update -qq",
    "apt-get install -qy libelf1",
    "npm install --global npm",
    "npm install --global yarn",
    "npm install",
  ],
  "test": Object {
    "image": "node:8",
    "script": Array [
      "npm test",
    ],
  },
}
`;
